easyedge-accelerometer-example/easy-edge-firmware/components/astarte_handler/astarteHandler.c

```diff
@@ -0,0 +1,214 @@
+#include "astarteHandler.h"
+#include "astarte_bson.h"
+#include "astarte_bson_types.h"
+#include "astarte_credentials.h"
+#include "astarte_hwid.h"
+#include "esp_log.h"
+#include "esp_wifi.h"
+#include "mbedtls/base64.h"
+#include <astarte_bson_serializer.h>
+#include <uuid.h>
+
+#define EXAMPLE_UUID "37119eb1-84fc-4e4b-97de-0b18ab1a49f1"
+#define MAC_LENGTH 6
+#define ENCODED_HWID_LENGTH 64
+#define MAC_STRING_LENGTH 13
+
+static const char *TAG = "ASTARTE_HANDLER";
+
+bool start(AstarteHandler *this);
+
+bool stop(AstarteHandler *this);
+
+void pub_acc_data(astarte_device_handle_t astarte_device, float x, float y, float z);
+
+const static astarte_interface_t acc_data_interface = { .name = "ai.clea.examples.Accelerometer",
+    .major_version = 0,
+    .minor_version = 1,
+    .ownership = OWNERSHIP_DEVICE,
+    .type = TYPE_DATASTREAM };
+
+const static astarte_interface_t server_datastream_interface = {
+    .name = "org.astarteplatform.esp32.ServerDatastream",
+    .major_version = 0,
+    .minor_version = 1,
+    .ownership = OWNERSHIP_SERVER,
+    .type = TYPE_DATASTREAM,
+};
+
+static void astarte_data_events_handler(astarte_device_data_event_t *event)
+{
+    ESP_LOGI(TAG, "Got Astarte data event, interface_name: %s, path: %s, bson_type: %d",
+        event->interface_name, event->path, event->bson_value_type);
+
+    if (strcmp(event->interface_name, server_datastream_interface.name) == 0
+        && strcmp(event->path, "/led") == 0 && event->bson_value_type == BSON_TYPE_BOOLEAN) {
+        int led_state = astarte_bson_value_to_int8(event->bson_value);
+
+        esp_event_post(ASTARTE_EVENTS, ASTARTE_EVENT_LED, (void*) &led_state, sizeof(int), 0);
+    }
+}
+
+static void astarte_connection_events_handler()
+{
+    esp_event_post(ASTARTE_EVENTS, ASTARTE_EVENT_CONNECT, NULL, 0, 0);
+    ESP_LOGI(TAG, "on_connected");
+}
+
+static void astarte_disconnection_events_handler()
+{
+    esp_event_post(ASTARTE_EVENTS, ASTARTE_EVENT_CONNECT, NULL, 0, 0);
+    ESP_LOGW(TAG, "on_disconnected");
+}
+
+esp_err_t add_interfaces(astarte_device_handle_t device)
+{
+    astarte_err_t ret;
+    ret = astarte_device_add_interface(device, &acc_data_interface);
+    ret |= astarte_device_add_interface(device, &server_datastream_interface);
+
+    if (ret == ASTARTE_OK) {
+        ESP_LOGI(TAG, "ADDING INTERFACES OK");
+        return ESP_OK;
+    } else {
+        ESP_LOGE(TAG, "ERROR ADDING INTERFACES");
+        return ESP_FAIL;
+    }
+}
+
+AstarteHandler *astarteHandler_init()
+{
+    AstarteHandler *astarteHandler = NULL;
+    astarte_device_handle_t device = NULL;
+    char *encoded_hwid = NULL;
+
+    astarte_credentials_use_nvs_storage(NVS_PARTITION);
+    astarte_credentials_init();
+
+    encoded_hwid = astarteHandler_getHardwareIdEncoded();
```

Comment:
C99 style is welcome, therefore: `char *encoded_hwid = astarteHandler_getHardwareIdEncoded();`